arangod/Aql/IndexNode.h

```diff
@@ -115,6 +116,31 @@ class IndexNode : public ExecutionNode, public DocumentProducingNode, public Col
   /// the projection attributes (if any)
   void initIndexCoversProjections();
 
+  void planNodeRegisters(std::vector<aql::RegisterId>& nrRegsHere,
+                         std::vector<aql::RegisterId>& nrRegs,
+                         std::unordered_map<aql::VariableId, aql::VarInfo>& varInfo,
+                         unsigned int& totalNrRegs, unsigned int depth) const;
+
+  bool isLateMaterialized() const noexcept {
+    TRI_ASSERT((_outNonMaterializedDocId == nullptr && _outNonMaterializedIndVars.empty()) ||
+               !(_outNonMaterializedDocId == nullptr || _outNonMaterializedIndVars.empty()));
+    return !_outNonMaterializedIndVars.empty();
+  }
+
+  struct IndexVariable {
+    TRI_idx_iid_t indexId;
+    size_t indexFieldNum;
+    Variable const* var;
+  };
+
+  using IndexValuesVars = std::unordered_map<TRI_idx_iid_t, std::unordered_map<size_t, Variable const*>>;
+
+  using IndexValuesRegisters = std::unordered_map<TRI_idx_iid_t, std::unordered_map<size_t, RegisterId>>;
```

Comment:
wouldn't be better to use 
```
std::unordered_map<TRI_idx_iid_t, std::map<size_t, RegisterId>>;
```
to avoid 
```
auto indReg = indRegs->second.find(i++);
```
in `getCallback<...>(...)`
???